generatebundlefile/ecr_test.go

```diff
@@ -0,0 +1,240 @@
+				t.Fatalf("splitECRName() = %#v\n\n\n, want %#v %#v", got, tc.chartName, tc.helmName)
+			}
+		})
+	}
+}
+
+//	err = UnTarHelmChart(chartPath, chartName, dest)
+
+func TestUnTarHelmChart(t *testing.T) {
+	tests := []struct {
+		testName      string
+		testChartPath string
+		testChartName string
+		dest          string
+		wantErr       bool
+	}{
+		{
+			testName:      "Test empty ChartPath",
+			testChartPath: "",
+			testChartName: "hello-eks-anywhere",
+			dest:          fmt.Sprintf("%s/go/src/github.com/modelrocket/eks-anywhere-packages/generatebundlefile/hello-eks-anywhere", os.Getenv("HOME")),
+			wantErr:       true,
+		},
+		{
+			testName:      "Test empty ChartName",
+			testChartPath: fmt.Sprintf("%s/Library/Caches/helm/repository/hello-eks-anywhere-0.1.1+9b09ef845d5d38d5201b96e32ae0be0ce2402b78.tgz", os.Getenv("HOME")),
+			testChartName: "",
+			dest:          fmt.Sprintf("%s/go/src/github.com/modelrocket/eks-anywhere-packages/generatebundlefile/hello-eks-anywhere", os.Getenv("HOME")),
+			wantErr:       true,
+		},
+		{
+			testName:      "Test empty dest",
+			testChartPath: fmt.Sprintf("%s/Library/Caches/helm/repository/hello-eks-anywhere-0.1.1+9b09ef845d5d38d5201b96e32ae0be0ce2402b78.tgz", os.Getenv("HOME")),
+			testChartName: "hello-eks-anywhere",
+			dest:          "",
+			wantErr:       true,
+		},
+		{
+			testName:      "Test valid values",
+			testChartPath: fmt.Sprintf("%s/Library/Caches/helm/repository/hello-eks-anywhere-0.1.1+9b09ef845d5d38d5201b96e32ae0be0ce2402b78.tgz", os.Getenv("HOME")),
+			testChartName: "hello-eks-anywhere",
+			dest:          fmt.Sprintf("%s/go/src/github.com/modelrocket/eks-anywhere-packages/generatebundlefile/hello-eks-anywhere", os.Getenv("HOME")),
+			wantErr:       false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.testName, func(tt *testing.T) {
+			err := UnTarHelmChart(tc.testChartPath, tc.testChartName, tc.dest)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("UnTarHelmChart() error = %v, wantErr %v", err, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestShaExistsInRepository(t *testing.T) {
+	tests := []struct {
+		testName       string
+		testRepository string
+		testVersion    string
+		checkPass      bool
+		wantErr        bool
+	}{
+		{
+			testName:       "Test empty Repository",
+			testRepository: "",
+			testVersion:    "sha256:0526725a65691944e831add6b247b25a93b8eeb1033dddadeaa089e95b021172",
+			wantErr:        true,
+		},
+		{
+			testName:       "Test empty Version",
+			testRepository: "hello-eks-anywhere",
+			testVersion:    "",
+			wantErr:        true,
+		},
+		{
+			testName:       "Test valid Repository and Version",
+			testRepository: "hello-eks-anywhere",
+			testVersion:    "sha256:0526725a65691944e831add6b247b25a93b8eeb1033dddadeaa089e95b021172",
+			checkPass:      true,
+			wantErr:        false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.testName, func(tt *testing.T) {
+			ecrPublicClient, err := NewECRPublicClient(true)
```

Comment:
This test is actually reaching out to AWS? That's not something we want to be doing. You should stub out the responses from AWS. We want our tests to run reliably and quickly.

Reply:
Added an issue for this.